base/task.jl

```diff
@@ -208,6 +225,7 @@ function produce(v)
         else
             schedule_and_wait(t, v)
         end
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
It's still very likely that there are other code path that are using this assumption and are not covered in your test.